jscomp/others/belt_MutableSet.mli

```diff
@@ -23,122 +23,589 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
-(** A _mutable_ sorted set module which allows customize _compare_ behavior.
-
-    Same as Belt.Set, but mutable.
+(**
+  A **mutable** sorted set module which allows customized compare behavior.
+  The implementation uses balanced binary trees, and therefore searching and insertion take time logarithmic in the size of the map.
+
+  It also has two specialized inner modules [Belt.MutableSet.Int](mutable-set-int) and [Belt.MutableSet.String](mutable-set-string) - This module separates data from function which is more verbose but slightly more efficient
+
+  ```res example
+  module PairComparator = Belt.Id.MakeComparable({
+    type t = (int, int)
+    let cmp = ((a0, a1), (b0, b1)) =>
+      switch Pervasives.compare(a0, b0) {
+      | 0 => Pervasives.compare(a1, b1)
+      | c => c
+      }
+  })
+
+  let mySet = Belt.MutableSet.make(~id=module(PairComparator))
+  mySet->Belt.MutableSet.add((1, 2))
+  ```
 *)
 
 
-(** Specalized when key type is `int`, more efficient
+(** Specialized when key type is `int`, more efficient
     than the generic type
 *)
 module Int = Belt_MutableSetInt
 
-(** Specalized when key type is `string`, more efficient
+(** Specialized when key type is `string`, more efficient
     than the generic type *)
 module String = Belt_MutableSetString
 
-type ('k,'id) t
+type ('value, 'identity) t
+(**
+  `'value` is the element type
 
-type ('k, 'id) id = ('k, 'id) Belt_Id.comparable
+  `'identity` the identity of the collection
+*)
+
+type ('value, 'id) id = ('value, 'id) Belt_Id.comparable
+(**
+  The identity needed for making a set from scratch
+*)
 
 val make: id:('value, 'id) id -> ('value, 'id) t
+(**
+  Creates a new set by taking in the comparator
+*)
+
+val fromArray: 'value array -> id:('value, 'id) id ->   ('value, 'id) t
+(**
+  Creates new set from array of elements.
+
+  ```res example
+  module IntCmp = Belt.Id.MakeComparable({
+    type t = int
+    let cmp = Pervasives.compare
+  })
+
+  let s0 = Belt.MutableSet.fromArray([1, 3, 2, 4], ~id=module(IntCmp))
+
+  s0->Belt.MutableSet.toArray /* [1, 2, 3, 4] */
+  ```
+*)
 
-val fromArray: 'k array -> id:('k, 'id) id ->   ('k, 'id) t
 val fromSortedArrayUnsafe: 'value array -> id:('value, 'id) id ->  ('value,'id) t
-val copy: ('k, 'id) t -> ('k, 'id) t
+(**
+  The same as [fromArray][#fromarray] except it is after assuming the input array is already sorted.
+*)
+
+val copy: ('value, 'id) t -> ('value, 'id) t
+(**
+  Returns copy of a set.
+
+  ```res example
+  module IntCmp = Belt.Id.MakeComparable({
+    type t = int
+    let cmp = Pervasives.compare
+  })
+
+  let s0 = Belt.MutableSet.fromArray([1, 3, 2, 4], ~id=module(IntCmp))
+
+  let copied = s0->Belt.MutableSet.copy
+  copied->Belt.MutableSet.toArray /* [1, 2, 3, 4] */
+  ```
+*)
+
 val isEmpty: _ t -> bool
```

Comment:
The underscore is just "ignoring" the type parameter of type `t` (most likely since we don't need to know the value types to figure out if a set is empty).

Maybe it's easier to read in ReScript syntax: `let isEmpty: (t<_>) => bool`